src/test/function_test/test_bulk_load.cpp

```diff
@@ -0,0 +1,254 @@
+        system("./run.sh start_onebox -w");
+        chdir(working_root_dir.c_str());
+    }
+
+public:
+    std::shared_ptr<replication_ddl_client> ddl_client;
+    pegasus::pegasus_client *pg_client;
+    std::string pegasus_root_dir;
+    std::string working_root_dir;
+    std::string bulk_load_local_root;
+    enum operation
+    {
+        GET,
+        SET,
+        DEL,
+        NO_VALUE
+    };
+
+public:
+    void copy_bulk_load_files()
+    {
+        chdir(pegasus_root_dir.c_str());
+        system("mkdir onebox/block_service");
+        system("mkdir onebox/block_service/local_service");
+        std::string copy_file_cmd = "cp -r src/test/function_test/bulk_load_files/" + LOCAL_ROOT +
+                                    " onebox/block_service/local_service";
+        system(copy_file_cmd.c_str());
+    }
+
+    error_code start_bulk_load()
+    {
+        auto err_resp = ddl_client->start_bulk_load(APP_NAME, CLUSTER, PROVIDER);
+        return err_resp.get_value().err;
+    }
+
+    void remove_file(const std::string &file_path)
+    {
+        std::string cmd = "rm " + file_path;
+        system(cmd.c_str());
+    }
+
+    void replace_bulk_load_info()
+    {
+        chdir(pegasus_root_dir.c_str());
+        std::string cmd = "cp -R src/test/function_test/bulk_load_files/mock_bulk_load_info/. " +
+                          bulk_load_local_root + "/" + CLUSTER + "/" + APP_NAME + "/";
+        system(cmd.c_str());
+    }
+
+    void wait_bulk_load_finish(int64_t seconds)
+    {
+        int64_t sleep_time = 5;
+        error_code err = ERR_OK;
+
+        while (seconds > 0 && err == ERR_OK) {
+            sleep_time = sleep_time > seconds ? seconds : sleep_time;
+            seconds -= sleep_time;
+            std::cout << "sleep " << sleep_time << "s to query bulk status" << std::endl;
+            std::this_thread::sleep_for(std::chrono::seconds(sleep_time));
+
+            err = ddl_client->query_bulk_load(APP_NAME).get_value().err;
+        }
+    }
+
+    bool verify_bulk_load_data()
+    {
+        if (!verify_data("hashkey", "sortkey")) {
+            return false;
+        }
+        return verify_data(HASHKEY_PREFIX, SORTKEY_PREFIX);
+    }
+
+    bool verify_data(const std::string &hashkey_prefix, const std::string &sortkey_prefix)
+    {
+        const std::string &expected_value = VALUE;
+        for (int i = 0; i < COUNT; ++i) {
+            std::string hash_key = hashkey_prefix + boost::lexical_cast<std::string>(i);
+            for (int j = 0; j < COUNT; ++j) {
+                std::string sort_key = sortkey_prefix + boost::lexical_cast<std::string>(j);
+                std::string act_value;
+                int ret = pg_client->get(hash_key, sort_key, act_value);
+                if (ret != 0) {
+                    std::cout << "Failed to get [" << hash_key << "," << sort_key << "], error is "
+                              << ret << std::endl;
+                    return false;
```

Comment:
Once this function return `false`, it will lead caller's `ASSERT_*` fail. So you can use ASSERT_* inside this function like:
```
ASSERT_EQ(ret, ERR_OK) << "Failed to get [" << hash_key << "," << sort_key << "], error is "
                       << ret << std::endl;
```

Reply:
I add assert statement in function `verify_bulk_load_data`